tests/integration/orcid/test_orcid_api.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""ORCID util tests."""
+
+from __future__ import absolute_import, division, print_function
+
+import os
+
+import pkg_resources
+import pytest
+import re
+import requests
+import requests_mock
+
+from inspirehep.modules.orcid.api import push_record_with_orcid, get_author_putcodes
+
+
+@pytest.fixture
+def mocked_internal_services(api_client):
+    hep_response = api_client.get('/literature/4328')
+    bibtex_response = api_client.get('/literature/4328', headers={
+        'Accept': 'application/x-bibtex',
+    })
+
+    def api_matcher(request):
+        if request.url == 'http://localhost:5000/api/literature/4328':
+            resp = requests.Response()
+            resp.status_code = 200
+            if request.headers['Accept'] == 'application/json':
+                resp._content = hep_response.data
+                return resp
+            elif request.headers['Accept'] == 'application/x-bibtex':
+                resp._content = bibtex_response.data
+                return resp
+
+    with requests_mock.Mocker() as requests_mocker:
+        requests_mocker.register_uri(
+            requests_mock.ANY,
+            re.compile('.*(indexer).*'),
+            real_http=True,
+        )
+        requests_mocker.add_matcher(api_matcher)
+        yield requests_mocker
+
+
+@pytest.fixture
+def mock_orcid_api(app, mocked_internal_services):
+    """Yield a MemberAPI and mock responses"""
+    mocked_internal_services.put(
+        'https://api.sandbox.orcid.org/v2.0/0000-0002-1825-0097/work/895497',
+        text='200 OK'
+    )
+    mocked_internal_services.post(
+        'https://api.sandbox.orcid.org/v2.0/0000-0002-1825-0097/work',
+        headers={
+            'Location': 'https://api.sandbox.orcid.org/v2.0/0000-0002-1825-0097/work/123456'
+        },
+    )
+    return mocked_internal_services
+
+
+def test_push_record_with_orcid(api_client, mock_orcid_api):
+    sample_orcid = '0000-0002-1825-0097'
+    expected_put_code = '895497'
+
+    result_put_code = push_record_with_orcid(
+        '4328', sample_orcid, 'fake-token', '895497'
```

Comment:
I would use the argument names in the call, it's not clear to me what is `'4328'` and what `'896597'`.

Reply:
btw. this one looks like it's just checking if push_record_with_orcid return the last of the parameters you passed to it xd